app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml

```diff
@@ -27,25 +28,18 @@ $paddingBottom = $block->getRatio() * 100;
 $styles = <<<STYLE
 .product-image-container-{$block->getProductId()} {
     width: {$width}px;
+    height: auto;
+    aspect-ratio: {$width} / {$height};
 }
```

Comment:
just curios if we don't have width or height declare or null somewhere ?

Reply:
some kind of fallback if width or height not set we should set aspect ratio to 1 ?